player/src/androidTest/kotlin/com/tidal/sdk/player/playlog/PlayLogTest.kt

```diff
@@ -203,6 +208,87 @@ class PlayLogTest {
+        loadAndPlayThenPauseThenPlay(MediaProduct(ProductType.TRACK, "1", null, null))
+
+    @Suppress("LongMethod")
+    private fun loadAndPlayThenPauseThenPlay(mediaProduct: MediaProduct) = runTest {
+        val gson = Gson()
+        responseDispatcher[
+            "https://api.tidal.com/v1/tracks/1/playbackinfo?playbackmode=STREAM&assetpresentation=FULL&audioquality=LOW".toHttpUrl(),
+        ] = {
+            MockResponse().setBodyFromFile(
+                "api-responses/playbackinfo/tracks/playlogtest/get_1_bts.json",
+            )
+        }
+        responseDispatcher["https://test.audio.tidal.com/1_bts.m4a".toHttpUrl()] = {
+            MockResponse().setBodyFromFile("raw/playlogtest/1_bts.m4a")
+        }
+
+        player.playbackEngine.load(mediaProduct)
+        player.playbackEngine.play()
+        withContext(Dispatchers.Default.limitedParallelism(1)) {
+            withTimeout(4.seconds) {
+                player.playbackEngine.events.filter { it is Event.MediaProductTransition }.first()
+            }
+            delay(2.seconds)
+            while (player.playbackEngine.assetPosition < 2) {
+                delay(10.milliseconds)
+            }
+            player.playbackEngine.pause()
+            delay(1.seconds)
+            player.playbackEngine.play()
+            withTimeout(8000) {
+                player.playbackEngine.events.filter { it is MediaProductEnded }.first()
+            }
+        }
+
+        eventReporterCoroutineScope.advanceUntilIdle()
+        verify(eventSender).sendEvent(
+            eq("playback_session"),
+            eq(ConsentCategory.NECESSARY),
+            argThat {
+                with(gson.fromJson(this, JsonObject::class.java)["payload"].asJsonObject) {
+                    assertThat(get("startAssetPosition").asDouble).isAssetPositionEqualTo(0.0)
+                    assertThat(get("endAssetPosition").asDouble).isAssetPositionEqualTo(5.065)
+                    assertThat(get("actualProductId").asString).isEqualTo(mediaProduct.productId)
+                    assertThat(get("sourceType")?.asString).isEqualTo(mediaProduct.sourceType)
+                    assertThat(get("sourceId")?.asString).isEqualTo(mediaProduct.sourceId)
+                    with(get("actions").asJsonArray) {
+                        val pauseAction =
+                            gson.fromJson(this[0], PlaybackSession.Payload.Action::class.java)
+                        assertThat(pauseAction.actionType)
+                            .isEqualTo(PlaybackSession.Payload.Action.Type.PLAYBACK_STOP)
+                        val resumeAction =
+                            gson.fromJson(this[1], PlaybackSession.Payload.Action::class.java)
+                        assertThat(resumeAction.actionType)
+                            .isEqualTo(PlaybackSession.Payload.Action.Type.PLAYBACK_START)
+                        assertThat(resumeAction.assetPositionSeconds)
```

Comment:
I think we should also link this up to the assetPosition we check for it to stop in the test itself: `player.playbackEngine.assetPosition < 2`. Something like `assertThat(resumeAction.assetPositionSeconds).isAssetPositionEqualTo(2)`. We don't just care about stop being similar to start, but that the value is actually what we expect it to be.

That said, we could do `assertThat(resumeAction.assetPositionSeconds).isEqualTo(pauseAction.assetPositionSeconds)` instead, right? Because we want them to be identical.

Reply:
I don't really understand the first paragraph, but your suggestion from the second one, I'm not sure. Or well, that would be probably correct in practice, but then again we know what happens with positions 0 and end of asset, so we don't know that things are going to be different for this case. So in that sense, they are positions just as the other ones so we should probably give them the same error margin I would say

Reply:
Yeah, I guess it's fine to have that leeway 👌 

For the first paragraph I just wanted two other assertions that checks that the assetPosition for both start and stop action is 2.